plugins/vpp/l3plugin/l3idx/l3_arp_index_test.go

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) 2018 Cisco and/or its affiliates.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package l3idx_test
+
+import (
+	"github.com/ligato/cn-infra/logging/logrus"
+	"github.com/ligato/vpp-agent/idxvpp"
+	"github.com/ligato/vpp-agent/idxvpp/nametoidx"
+	"github.com/ligato/vpp-agent/plugins/vpp/l3plugin/l3idx"
+	"github.com/ligato/vpp-agent/plugins/vpp/model/l3"
+	. "github.com/onsi/gomega"
+	"testing"
+)
+
+func l3arpIndexTestInitialization(t *testing.T) (idxvpp.NameToIdxRW, l3idx.ARPIndexRW) {
+	RegisterTestingT(t)
+
+	// initialize index
+	nameToIdx := nametoidx.NewNameToIdx(logrus.DefaultLogger(), "index_test", nil)
+	index := l3idx.NewARPIndex(nameToIdx)
+	names := nameToIdx.ListNames()
+
+	// check if names were empty
+	Expect(names).To(BeEmpty())
+
+	return index.GetMapping(), index
+}
+
+var arpEntries = []l3.ArpTable_ArpEntry{
+	{
+		Interface:   "tap1",
+		IpAddress:   "192.168.10.21",
+		PhysAddress: "59:6C:DE:AD:00:01",
+		Static:      true,
+	},
+	{
+		Interface:   "tap2",
+		IpAddress:   "192.168.10.22",
+		PhysAddress: "59:6C:DE:AD:00:02",
+		Static:      true,
+	},
+	{
+		Interface:   "tap3",
+		IpAddress:   "dead::01",
+		PhysAddress: "59:6C:DE:AD:00:03",
+		Static:      false,
+	},
+}
+
+func TestArpRegisterAndUnregisterName(t *testing.T) {
+	mapping, l3index := l3arpIndexTestInitialization(t)
+
+	// Register entry
+	l3index.RegisterName("l3", 0, &arpEntries[0])
+	names := mapping.ListNames()
+	Expect(names).To(HaveLen(1))
+	Expect(names).To(ContainElement("l3"))
+
+	// Unregister entry
+	l3index.UnregisterName("l3")
+	names = mapping.ListNames()
+	Expect(names).To(BeEmpty())
+}
+
+func TestArpLookupIndex(t *testing.T) {
+	RegisterTestingT(t)
```

Comment:
Unnecessary, already called in `l3arpIndexTestInitialization`.

Reply:
LookupIndex is not called in l3arpIndexTestInitialization. Or ... ?

Reply:
The point is that you are calling RegisterTesting(t) here and also in initialization method l3arpIndexTestInitialization below